uptane/services/inventorydb.py

```diff
@@ -84,122 +84,94 @@ def scrub_filename(fname, expected_containing_dir):
-  def save_ecu_manifest(self, ecu_serial, signed_ecu_manifest):
-    uptane.formats.VIN_SCHEMA.check_match(vin)
-    uptane.formats.ECU_SERIAL_SCHEMA.check_match(ecu_serial)
-    uptane.formats.SIGNABLE_ECU_VERSION_MANIFEST_SCHEMA.check_match(
-        signed_ecu_manifest)
-
-    ecu_serial_ids = self.inventory_db['ecu_manifests']
-    if (ecu_serials_ids and (ecu_serial in ecu_serials_ids)):
-      (self.inventory_db['ecu_manifests'][ecu_serial]).append(signed_vehicle_manifest)
-    else:
-      ecu_list = []
-      ecu_list.append(signed_vehicle_manifest)
-      ecu_serial_ids[ecu_serial] = ecu_list
-      self.inventory_db['ecu_manifests'] = ecu_serial_ids
 
+# Global dictionaries
+vehicle_manifests_dic = {'vehicle_manifests': {}}
+ecu_manifests_dic = {'ecu_manifests': {}}
+primary_public_keys = {'vehicle_primaries': {}}
+public_keys = {'all_ecus': {}}
+
+
+
+# Save ECU manifest
+def save_ecu_manifest(ecu_serial, signed_ecu_manifest):
+
+  uptane.formats.VIN_SCHEMA.check_match(vin)
+  uptane.formats.ECU_SERIAL_SCHEMA.check_match(ecu_serial)
+  uptane.formats.SIGNABLE_ECU_VERSION_MANIFEST_SCHEMA.check_match(
+       signed_ecu_manifest)
 
-  # Get ECU manifest
-  def get_ecu_manifest(self, ecu_serial):
-    # Check arg format
-    ecu_serial_ids = self.inventory_db['ecu_manifests']
-    if (ecu_serial_ids and (ecu_serial in ecu_serial__ids)):
-      return self.inventory_db['ecu_manifests'][ecu_serial]
-    return False
+  if ecu_serial not in public_keys['all_ecus']:
+    raise uptane.UnknownECU("ECU serial has not been registered yet!")
+
+  if ecu_serial not in ecu_manifests_dic['vehicle_manifests']:
+    ecu_list = []
+    ecu_list.append(signed_ecu_manifest)
+    ecu_manifests_dic['ecu_manifests'][ecu_serial] = ecu_list
+  else:
+    ecu_manifests_dic['ecu_manifests'][ecu_serial].append(signed_vehicle_manifest)
+
+
+
+# Get ECU manifest
+def get_ecu_manifest(vin):
+
+  uptane.formats.VIN_SCHEMA.check_match(vin)
+
+  if vin not in ecu_manifests_dic['ecu_manifests']:
+    raise uptane.Error('The given VIN, ' + repr(vin) + ', is not known.')
+  else:
+    return ecu_manifests_dic['ecu_manifests'][vin]
+
+
+# Save vehicle manifest
+def save_vehicle_manifest(vin, signed_vehicle_manifest):
+
+  uptane.formats.VIN_SCHEMA.check_match(vin)
+  uptane.formats.SIGNABLE_ECU_VERSION_MANIFEST_SCHEMA.check_match(
+       signed_ecu_manifest)
+
+  if vin not in vehicle_manifests_dic['vehicle_manifests']:
+    vm_list = []
+    vm_list.append(signed_vehicle_manifest)
+    vehicle_manifests_dic['vehicle_manifests'][vin] = vm_list
+  else:
+    vehicle_manifests_dic['vehicle_manifests'][vin].append(signed_vehicle_manifest)
+
+  # Save all the contained ECU manifests.
+  # NOTE: not tested
+  all_contained_ecu_manifests = signed_vehicle_manifest['signed']['ecu_manifests']
+  for ecu_serial in all_contained_ecu_manifests:
+    save_ecu_manifest(ecu_serial, all_contained_ecu_manifests[ecu_serial])
+
+
+# Get vehicle manifest
+def get_vehicle_manifest(vin):
```

Comment:
This comment has not been addressed:
https://github.com/awwad/uptane/pull/21#discussion_r91142337

It is the same for get_ecu_manifest.